### PR TITLE
Enable UCC/cloudadapter flow for Windows

### PR DESCRIPTION
**Pull request by gblewis1**

### CODE REVIEW

**Comment by gblewis1** [comment]
@nmgaston @tsirlapu please review

File: inbm/cloudadapter-agent/cloudadapter/cloud/adapters/generic_adapter.py [hunk 1]
```diff
@@ -12,12 +12,13 @@
 from typing import Callable
 from ..client.cloud_client import CloudClient
 from .adapter import Adapter
-
+import snoop
```
Comment (gblewis1) [inline inbm/cloudadapter-agent/cloudadapter/cloud/adapters/generic_adapter.py:15]: remove snoop debugging

File: inbm/cloudadapter-agent/cloudadapter/cloud/adapters/generic_adapter.py [hunk 2]
```diff
@@ -12,12 +12,13 @@
...
 
 class GenericAdapter(Adapter):
     def __init__(self, config: dict):
         super().__init__(config)
 
+    @snoop
```
Comment (gblewis1) [inline inbm/cloudadapter-agent/cloudadapter/cloud/adapters/generic_adapter.py:21]: remove snoop debugging

File: inbm-lib/setup.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
     license='Intel Proprietary',
     packages=['inbm_lib', 'inbm_common_lib'],
     include_package_data=True,
-    install_requires=['paho-mqtt==1.5.1', 'dmidecode==0.9.0', 'xmlschema==1.5.3', 'defusedxml==0.7.1', 'future==0.18.3', 'url-normalize==1.4.3'],
+    install_requires=['paho-mqtt==1.5.1', 'dmidecode==0.9.0', 'xmlschema==1.5.3', 'defusedxml==0.7.1', 'future==0.18.3', 'url-normalize==1.4.3', 'snoop==0.4.3'],
```
Comment (gblewis1) [inline inbm-lib/setup.py:18]: remove snoop

File: inbm/cloudadapter-agent/cloudadapter/cloud/cloud_builders.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 from typing import Dict, Any, Optional
 import jsonschema
 import json
+import snoop
```
Comment (gblewis1) [inline inbm/cloudadapter-agent/cloudadapter/cloud/cloud_builders.py:20]: remove snoop debugging

File: inbm/cloudadapter-agent/cloudadapter/cloud/cloud_builders.py [hunk 5]
```diff
@@ -32,6 +33,7 @@ def validate_config(config: Dict[str, Any]) -> None:
         jsonschema.validate(config, schema=schema)
 
 
+@snoop(depth=2)
```
Comment (gblewis1) [inline inbm/cloudadapter-agent/cloudadapter/cloud/cloud_builders.py:36]: remove snoop debugging

File: inbm/cloudadapter-agent/cloudadapter/cloudadapter.py [hunk 6]
```diff
@@ -57,6 +58,7 @@ def start(self) -> None:
         logger.info('Cloud Adapter agent is running')
 
         # Exit if configuration is malformed
+        snoop.install(out=logger.info)
```
Comment (gblewis1) [inline inbm/cloudadapter-agent/cloudadapter/cloudadapter.py:61]: remove snoop debugging

File: inbm/fpm/inb-provision-cloud/main.go [hunk 7]
```diff
@@ -38,6 +37,28 @@ func main() {
 	flag.Usage = usage
 	flag.Parse()
 
+	var uccClientIdFile string
+	var uccServerIdFile string
+	var cloudCredentialPublicDir string
+	var cloudCredentialSecretDir string
+
+	if runtime.GOOS == "windows" {
+		uccClientIdFile = "c:\\intel-manageability\\inbm\\etc\\ucc\\client_id"
```
Comment (gblewis1) [inline inbm/fpm/inb-provision-cloud/main.go:46]: is this where UCC wants their server_id, client_id in Windows? Will they have a different path?

File: inbm/packaging/fake-dbs/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+FROM registry.hub.docker.com/library/bash:5.0.7
```
Comment (gblewis1) [inline inbm/packaging/fake-dbs/Dockerfile:1]: oops. remove fake-dbs directory.

**Comment by nmgaston** [comment]
I assume we'll be adding 'snoop' to our protex?

**Comment by gblewis1** [comment]
> I assume we'll be adding 'snoop' to our protex and OSPDT?

I removed snoop completely just now. I missed a couple references.

**Review by nmgaston — APPROVED** [review]

**Comment by gblewis1** [comment]
Ran Windows integration test, passed.

**Comment by gblewis1** [comment]
NOTE: leaving snoop in for the time being for debugging. If this gets merged with snoop, we need to add snoop (MIT license) to our BOM. Alternatively we can remove snoop and merge without changing the BOM.

File: inbm/fpm/inb-provision-cloud/main.go [hunk 9]
```diff
@@ -37,6 +36,28 @@ func main() {
 	flag.Usage = usage
 	flag.Parse()
 
+	var uccClientIdFile string
+	var uccServerIdFile string
+	var cloudCredentialPublicDir string
+	var cloudCredentialSecretDir string
+
+	if runtime.GOOS == "windows" {
+		uccClientIdFile = "c:\\ucc\\etc\\client_id"
```
Comment (pillaiabhishek) [inline inbm/fpm/inb-provision-cloud/main.go:45]: Maybe drop the "etc" in the windows path, i.e. c:\ucc\client_id and c:\ucc\server_id

Comment (gblewis1) [reply inbm/fpm/inb-provision-cloud/main.go]: > Maybe drop the "etc" in the windows path, i.e. c:\ucc\client_id and c:\ucc\server_id

good idea.. i've changed it.

**Pull request merged**